src/providers/intuit.ts

```diff
@@ -0,0 +1,79 @@
+import { OAuth2Client } from "oslo/oauth2";
+import { TimeSpan, createDate } from "oslo";
+
+import type { OAuth2Provider } from "../index.js";
+
+const authorizeEndpoint = "https://appcenter.intuit.com/connect/oauth2";
+const tokenEndpoint = "https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer";
+
+export class Intuit implements OAuth2Provider {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(clientId: string, clientSecret: string, redirectURI: string) {
+		this.client = new OAuth2Client(clientId, authorizeEndpoint, tokenEndpoint, {
+			redirectURI
+		});
+		this.clientSecret = clientSecret;
+	}
+
+	public async createAuthorizationURL(
+		state: string,
+		options?: {
+			scopes?: string[];
+		}
+	): Promise<URL> {
+		const scopes = options?.scopes ?? [];
+		return await this.client.createAuthorizationURL({
+			state,
+			scopes: [...scopes, "openid"]
+		});
+	}
+
+	public async validateAuthorizationCode(code: string): Promise<IntuitTokens> {
+		const result = await this.client.validateAuthorizationCode<TokenResponseBody>(code, {
+			authenticateWith: "request_body",
```

Comment:
The credentials should be send within the authorization header, so this should be `http_basic_auth`. That's the default in Oslo, you can just remove this line.

https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0#exchange-code-for-tokens